urdf/sdf_extensions.md

```diff
@@ -0,0 +1,822 @@
+# SDFormat extensions to URDF (the `<gazebo>` tag)
+
+When URDF files are loaded by Gazebo (both Gazebo-classic and the new Gazebo),
+the URDF content is first converted to SDFormat before being processed by
+Gazebo. Users who to modify the resulting SDFormat output to include SDFormat
+specific elements may do so using the `<gazebo>` tag in the original URDF file.
+This is known as the Gazebo extension to the [URDF
+specification](http://wiki.ros.org/urdf/XML) (see
+http://wiki.ros.org/urdf/XML/Gazebo)
+
+The URDF to SDFormat conversion usually happens automatically without users
+observing the resultang SDFormat file. To diagnose any issues that might come
+up during the conversion, the tool 
+
+```gz sdf -p <path to urdf file>```
+
+can be used to convert the URDF file to SDFormat that can be inspected by the user.
+
+The following is the documentation of the various tags available under
+`<gazebo>` and their corresponding effect on the SDFormat output. While some of
+the content included here is also available in the ["Using a URDF in
+Gazebo"](http://gazebosim.org/tutorials?tut=ros_urdf&cat=connect_ros) tutorial
+of Gazebo-classic, this document should provide a more complete discussion of
+the extension, especially regarding fixed joint lumping.
+
+
+## `<gazebo>` Elements For `<robot>`
+`<gazebo>` tag without a `reference` attribute applies to the SDFormat `<model>`
+that gets generated from the `<robot>` URDF tag. All elements in the `<gazebo>`
+tag are inserted into the SDF `<model>` tag for the generated SDF.
+
+**Example:**
+
+```xml
+<!--URDF-->
+<robot name='no_ref_example'>
+  <link name='world'/>
+  <gazebo>
+    <static>true</static>
+    <plugin name='testPlugin' filename='testFileName'/>
+  </gazebo>
+</robot>
```

Comment:
how do you feel about using `<include/>` elements in our Markdown files instead of duplicating the example code?

for example:

* https://classic.gazebosim.org/tutorials?tut=drcsim_animate_joints&cat=drcsim#ModelPluginController
* https://github.com/osrf/gazebo_tutorials/blame/0059020f4b86ae8f736104ef8f6c164f8830bd37/drcsim_animate_joints/tutorial.md#L46

I believe sdformat.org supports the same syntax, though I guess the links may have to point to your fork until the pull request is merged

Reply:
Thanks for the pointer. I was hoping there was a way to do that, but I didn't see any examples in sdf_tutorials. I've updated all the examples. https://github.com/gazebosim/sdf_tutorials/pull/88/commits/3beee03fc26bf4c00d3002e39380d7672c7351ca